scripts/categorizer.py

```diff
@@ -0,0 +1,59 @@
+
+
+def find_empty_lines(line):
+    """
+    Returns 1 for an empty line and 0 for non-empty
+    :param line:
+    :return:
+    """
+    if line.strip("\t").strip(" ").strip("\n") == '':
+        return 1
+    else:
+        return 0
+
+
+
+def main():
+    """
+    Currently this function does some prescripted stuff.
+
+    :return:
+    """
+    lines = []
+    with open("./categorizer.py") as fp:
+        lines = fp.readlines()
+
+    empty_lines = []
```

Comment:
Consider using different data structure like a named tuple as per chris's suggestion....

Reply:
I agree with packaging everything in a single packet, but not as I'm working with it as this keeps it simplier for me at this point. I will package them at the end of the categorizations to be ("file name: line_number", "assignment_boolean", "conditional_boolean", etc.). I find it easier to reason about for me when it is lists. It also makes it trivially parrallelizable keeping lists if needed. just pass the lines to several functions as read then have the functions separately write to their individual write memories in each list that is sequentially rather than all over the heap in case one categorizer goes faster than the others.. in a single data structure

Reply:
The point is one single data structure being sent to another program with json should be the end product, but I disagree with it being the intermediate products that I work with.